src/main/java/org/spongepowered/common/entity/player/SpongeUser.java

```diff
@@ -316,4 +313,82 @@ public void save() {
         }
     }
 
+    // Helpers for UserInventory
+
+    private <T> T getForInventory(Function<Player, T> playerFunction, Function<SpongeUser, T> userFunction)
+    {
+        if (this.self.getPlayer().isPresent()) {
+            return playerFunction.apply(this.self.getPlayer().get());
+        }
+        return userFunction.apply(this.loadInventory()); // Load Inventory if not yet loaded
+    }
+
+    private void setForInventory(Consumer<Player> playerFunction, Consumer<SpongeUser> userFunction)
+    {
+        if (this.self.getPlayer().isPresent()) {
+            playerFunction.accept(this.self.getPlayer().get());
+            return;
+        }
+        userFunction.accept(this.loadInventory()); // Load Inventory if not yet loaded
+    }
+
+
+    // Helpers for Equipment:
+
+    private Optional<ItemStack> getEquippedItem(EquipmentType type) {
+        if (type instanceof SpongeEquipmentType) {
+            EntityEquipmentSlot[] slots = ((SpongeEquipmentType) type).getSlots();
+            if (slots.length == 1) {
+                net.minecraft.item.ItemStack nmsItem = this.getItemStackFromSlot(slots[0]);
+                if (!nmsItem.isEmpty()) {
+                    return Optional.of(ItemStackUtil.fromNative(nmsItem));
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private void setEquippedItem(EquipmentType type, ItemStack item) {
+        if (type instanceof SpongeEquipmentType) {
+            EntityEquipmentSlot[] slots = ((SpongeEquipmentType) type).getSlots();
+            for (EntityEquipmentSlot slot : slots) {
+                this.setItemStackToSlot(slot, ItemStackUtil.toNative(item));
+                // TODO check canequip
+                return;
+            }
+        }
+    }
+
+    private net.minecraft.item.ItemStack getItemStackFromSlot(EntityEquipmentSlot slotIn)
+    {
+        if (slotIn == EntityEquipmentSlot.MAINHAND)
+        {
+            return this.inventory.getCurrentItem();
+        }
+        else if (slotIn == EntityEquipmentSlot.OFFHAND)
+        {
+            return this.inventory.offHandInventory.get(0);
+        }
+        else
+        {
+            return slotIn.getSlotType() == EntityEquipmentSlot.Type.ARMOR ? this.inventory.armorInventory.get(slotIn.getIndex()) : net.minecraft.item.ItemStack.EMPTY;
```

Comment:
You could multiline this ternary to avoid issues with figuring out where an error occurs in the future (if there are any that occur in the future)